### PR TITLE
Adding Logical DHCP port resource

### PR DESCRIPTION
**Pull request by asarfaty**
This resource will create a logical port on a logical switch with
an attachment to a DHCP server.
In addition - Fixing and improving the logical dhcp server resource and tests.

### CODE REVIEW

File: nsxt/resource_nsxt_logical_dhcp_port.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			"logical_switch_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Id of the Logical switch that this port belongs to",
+				Required:    true,
+				ForceNew:    true, // Cannot change the logical switch of a logical port
+			},
+			"dhcp_server_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Id of the Logical DHCP server this port belongs to",
+				Required:    true,
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_dhcp_port.go]: This one is not forcenew?

File: nsxt/resource_nsxt_logical_dhcp_port.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+)
+
+func resourceNsxtLogicalDhcpPort() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtLogicalDhcpPortCreate,
+		Read:   resourceNsxtLogicalDhcpPortRead,
+		Update: resourceNsxtLogicalDhcpPortUpdate,
+		Delete: resourceNsxtLogicalDhcpPortDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_dhcp_port.go]: Perhaps we should verify this port has dhcp attachment on import (read below only verifies attachment exists)

File: nsxt/resource_nsxt_logical_dhcp_server_test.go [hunk 3]
```diff
@@ -228,3 +282,44 @@ resource "nsxt_logical_dhcp_server" "test" {
...
+    scope = "scope1"
+    tag   = "tag1"
+  }
+}`, name, ip1, ip2, ip3)
+}
+
+func testAccNSXLogicalDhcpServerUpdateNoOptsTemplate(edgeClusterName string, updatedName string, ip1 string, ip2 string, ip3 string, ip4 string) string {
+	return testAccNSXDhcpServerProfileCreateForServerTemplate(edgeClusterName) + fmt.Sprintf(`
+resource "nsxt_logical_dhcp_server" "test" {
+  display_name                = "%s"
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_dhcp_server_test.go]: nit: formatting

File: nsxt/resource_nsxt_logical_dhcp_port.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_dhcp_port.go]: No. Changing it is supported by the NSX

File: nsxt/resource_nsxt_logical_dhcp_port.go (hunk 2 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_dhcp_port.go]: Thanks. I'll add this check to the Read function

File: nsxt/resource_nsxt_logical_dhcp_server_test.go (hunk 3 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_dhcp_server_test.go]: Done

**Pull request merged**